chore(puppeteer): remove puppeteer and related files (#2843)
toptal/picasso: docs/contribution/visual-testing.md

# Visual Testing

To maintain stability between versions we’ve integrated snapshotting tool to browse storybook of all components, creating screenshot of the page and comparing the diff with [`happo`](https://happo.io).

> It walks through all components and check all `story/index.jsx` files, read it as AST transformation tree and tries to generate all necessary examples which need to be saved as screenshots.

Reviewer: ```suggestion
It goes through all components and checks all `story/index.jsx` files, reads it as AST transformation tree and tries to generate all necessary examples which need to be saved as screenshots.
```

> It walks through all components and check all `story/index.jsx` files, read it as AST transformation tree and tries to generate all necessary examples which need to be saved as screenshots.

Reviewer 2: There's no AST crazyness here. You can just remove this, mostly, and replace it with Happo's Storybook plugin documentation link.

Reviewer 2: > generate all necessary examples which need to be saved as screenshots.

This doesn't happen btw. Nothing is saved, it's just in the cloud, in Happo.

> Storing/Updating snapshots should be handled along with `happo.io`. Happo stores the `correct` and `latest` state of the components from master/main branch. If during implementation there is any visual change, happo will reports this error to a report and Developer needs to manually check if the current state is expected, then developer approves current state as expected one.

Reviewer: ```suggestion
Storing/Updating snapshots should be handled along with `happo.io`. Happo stores the `correct` and `latest` state of the components from master/main branch. If during implementation there is any visual change, happo will report this error and Developer needs to manually check if the current state is expected, then developer approves current state as expected one.
```

## Fixing broken visual tests inside a PR

In order to have a good tracking on visual changes which were applied in each PR, we run visual snapshot comparison inside each pull request. `Happo` is automatically trying to take snapshot of each story and compare the result to a previous state. When your PR has failing visual tests you need to update them.

1. Check the report on `happo.io` which are linked to the PR status.
2. Manually check the differences by eye and ensure that the current state of the screenshot is the expected state.

> 3. After the engineer is sure that the current changes are legitimate approve them on that specific `happo.io` report.

Reviewer: ```suggestion
3. After the engineer is sure that the current changes are legitimate, approve them on that specific `happo.io` report.
```
